### DISCUSSION TITLE
Touch Input method for osu!mania

### DISCUSSION THREAD

**Discussion by paring-chan** [ppy/osu#16267] [Ideas] [answered]
I'm working on touch inputs for osu!mania ruleset.
But I don't think touching column is not a good idea. The column is too narrow to touch.
So I want to collect ideas for that.

**Comment by bdach** [comment, marked as answer]
see existing discussion at: https://github.com/ppy/osu/issues/4201, https://github.com/ppy/osu/issues/4195 and also the [osu!dev discord](https://discord.com/channels/188630481301012481/188630652340404224/924993320093098015).

clicking the columns may be fine (although IMO the *entire* column shouldn't accept input, just the actual key area), but a virtual keyboard with a larger size may be advisable.

**Reply by paring-chan** [reply]
![image](https://user-images.githubusercontent.com/68010770/147565727-e0d0818b-7918-4981-84ce-dadeef9a662b.png)
Do you mean this area?

**Reply by bdach** [reply]
yes

**Reply by smoogipoo** [reply]
What's your reasoning for only accepting input in the key area?

**Reply by bdach** [reply]
Well it instinctively makes sense to me to only record touches in the key area since that's the area that lights up if you normally press a keyboard key. It's pretty weird to allow touching on the entire column; each column above the key area is "dead space", as in you can't touch the actual notes there, you control only whether a column is pressed or not. Clicking all over the column would be to the player's detriment because they'd be obscuring their own field of view of incoming notes with their fingers.

**Reply by peppy** [reply]
Personally I'm not against making the whole column tappable. If anything, would handle the odd case a user touches slightly too high.

**Reply by smoogipoo** [reply]
Yeah, that was my thinking also. I think if the player's playing to their own detriment then that's a them problem.

**Reply by bdach** [reply]
I suppose you could also have some tolerance on the key area to make mistouches less possible, but at the point tolerance is being added then it's probably better and easier to just use the entire column, so sure I suppose.
